src/s1reader/s1_etad.py

```diff
@@ -0,0 +1,202 @@
+        eta_az_start = (eta.min_azimuth_time - slc_burst.sensing_mid).total_seconds() + eta_burst.sampling_start['y']
+        eta_rg_start = eta.min_range_time + eta_burst.sampling_start['x']
+        eta_az_ax = eta_az_start + np.arange(eta_burst.lines) * eta_burst.sampling['y']
+        eta_rg_ax = eta_rg_start + np.arange(eta_burst.samples) * eta_burst.sampling['x']
+
+        # calculate SLC grid
+        slc_az_ax = np.arange(slc_burst.length) * slc_burst.azimuth_time_interval \
+                    + (slc_burst.sensing_start - slc_burst.sensing_mid).total_seconds()
+        slc_rg_ax = np.arange(slc_burst.width) / slc_burst.range_sampling_rate \
+                    + slc_burst.slant_range_time
+
+    if resample:
+        # resample ETA correction data to the SLC grid
+        vprint('resampling the ETA correction data from ETA grid to SLC grid ...')
+        rg_interp = RectBivariateSpline(eta_az_ax, eta_rg_ax, eta_rg_corr, kx=1, ky=1)  # bi-linear
+        az_interp = RectBivariateSpline(eta_az_ax, eta_rg_ax, eta_az_corr, kx=1, ky=1)  # bi-linear
+        slc_rg_corr = rg_interp(slc_az_ax, slc_rg_ax)
+        slc_az_corr = az_interp(slc_az_ax, slc_rg_ax)
+
+    if plot:
+        vprint('plot ETA correction data and grid')
+
+        # figure 1 - ETA corrections
+        fig, axs = plt.subplots(nrows=2, ncols=1, figsize=[12, 6], sharex=True, sharey=True)
+        for ax, corr, title in zip(axs, [eta_rg_corr, eta_az_corr], ['x', 'y']):
+            im = ax.imshow(corr, aspect='auto', interpolation='nearest')
+            fig.colorbar(im, ax=ax, shrink=0.8, location='right').set_label('pixel')
+            ax.set_title(f'correction [{title}]')
+            ax.set_ylabel('Azimuth [pixel]')
+        axs[1].set_xlabel('Range [pixel]')
+        fig.tight_layout()
+
+        # figure 2 - ETA & SLC grids [for comparison/checking]
+        eta_box = np.asarray([
+            (eta_rg_ax[0],  eta_az_ax[0]),
+            (eta_rg_ax[0],  eta_az_ax[-1]),
+            (eta_rg_ax[-1], eta_az_ax[-1]),
+            (eta_rg_ax[-1], eta_az_ax[0]),
+            (eta_rg_ax[0],  eta_az_ax[0]),
+        ])
+        slc_box = np.asarray([
+            (slc_rg_ax[0],  slc_az_ax[0]),
+            (slc_rg_ax[0],  slc_az_ax[-1]),
+            (slc_rg_ax[-1], slc_az_ax[-1]),
+            (slc_rg_ax[-1], slc_az_ax[0]),
+            (slc_rg_ax[0],  slc_az_ax[0]),
+        ])
+
+        fig, ax = plt.subplots(figsize=[12, 3])
+        ax.plot(eta_box[:, 0] * 1e3, eta_box[:, 1], 'C0', label='ETA')
+        ax.plot(slc_box[:, 0] * 1e3, slc_box[:, 1], 'C1', label='SLC')
+        ax.set_xlabel('Range [ms]')
+        ax.set_ylabel('Azimuth [s]')
+        ax.set_title('grid')
+        ax.legend()
+        ax.grid()
+
+        plt.show()
+
+    if not resample:
+        return eta_rg_corr, eta_az_corr
+
+    return slc_rg_corr, slc_az_corr
+
+
+
+def get_eta_burst_from_slc_burst(slc_burst, eta_dir, verbose=True):
+    """Read ETA burst corresponding to the input SLC burst."""
```

Comment:
What does ETA stand for?

Reply:
ETA stands for extended timing annotation, it's a term used in the SAFE file/dir by ESA, similar to SLC, e.g. `S1A_IW_ETA__AXSV_20160805T230627_20160805T230654_012471_0137E2_91C6.SAFE`

Reply:
I see, I was confused because I have always heard ETAD. Thanks for the clarification.